mlflow/entities/trace.py

```diff
@@ -100,6 +105,88 @@ def to_pandas_dataframe_row(self) -> dict[str, Any]:
             "tags": self.info.tags,
         }
 
+    def search_spans(
+        self, span_type: Optional[SpanType] = None, name: Optional[Union[str, Pattern]] = None
```

Comment:
```suggestion
        self, span_type: Optional[SpanType] = None, name: Optional[Union[str, re.Pattern]] = None
```

https://docs.python.org/3/library/typing.html#typing.Pattern is deprecated.